### ISSUE TITLE
Require label error is triggered when label isn't set on the first stage and set on the next for multistage Dockerfile

### ISSUE THREAD

**Issue by Jasstkn** [hadolint/hadolint#758]
- [x] This is a bug report
- [ ] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behavior

No errors. There is no need to copy paste LABEL section because at the end only the final stage will be saved.

### Actual behavior

Errors about missing label

```bash
Dockerfile:1 DL3049 info: Label `team` is missing.
Dockerfile:1 DL3049 info: Label `maintainer` is missing.
```

### Steps to reproduce the behavior

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version` or
  `docker run --rm ghcr.io/hadolint/hadolint hadolint --version`:**

```bash
Haskell Dockerfile Linter 2.8.0
```

**Dockerfile** (if relevant)

```Dockerfile
FROM alpine:3.11 as build

....

FROM alpine:3.11

LABEL team="devops" maintainer="example.com"

.....

```

**Additional environment details (OS, stack version, etc.)**

**Comment by m-ildefons** [comment]
Hey,
sorry I sat on this so long, but life got in the way...
Anyways, I think the PR is a reasonable improvement over the previous behavior. The example Dockerfile above will still trigger DL3049, because the latter stage doesn't inherit from the first one, but this Dockerfile for example should be completely fine, as one would expect:
```Dockerfile
FROM opensuse/leap:15.4 AS builder
...
FROM builder
LABEL team="devops" maintainer="example.com"
```
